### ISSUE TITLE
Enable Theming by default

### ISSUE THREAD

**Issue by Pilzinsel64** [nextcloud-snap/nextcloud-snap#2456]
As I regoniced during #2452: What do you think should we do with the thiming app? In Nextcloud since 26 (or even 25?) it is a hard dependency and you can't even disable the app on the normal way with the message "This app can not be disabled.". It isn't even listed on the Web Interface as installable app anymore, like the files app e.g.

Should we go ahead and also enable theming for snap by default? I mean, it is a core feature and were an big feature for the NC25 update and is also used for accessibility, if I'm right. Not sure if keep it disabled is the best anymore. I mean, if someone has problems due theming we can navigate him to the the upstream repo, same as with other core issues. What do you think? 🤔 

I opened an issue for to discuss that. Maybe we can share our experiences with. On my side it mostly works fine.

_Originally posted by @Pilzinsel64 in https://github.com/nextcloud-snap/nextcloud-snap/issues/2452#issuecomment-1632975067_

**Comment by scubamuc** [comment]
> I mean, it is a core feature and were an big feature for the NC25 update and is also used for accessibility, if I'm right. Not sure if keep it disabled is the best anymore. 

> Theming mostly works fine on my systems, sometimes needs reset and reconfigure after upgrade.

agree, core features should be enabled by default.

**Comment by Pilzinsel64** [comment]
> Theming mostly works fine on my systems, sometimes needs reset and reconfigure after upgrade.

True. I remember now, NC24 were the big theming update. This broke everything. Also previous version sometimes did reset the settings, so I needed to re-configure coloure, background and favicons.
Since NC25 I had no problems anymore. Maybe they fixed it, but I didn't find a pull request.

**Comment by scubamuc** [comment]
> Since NC25 I had no problems anymore. 

one of my instances needed a reconfiguration after uprade to 26.0.3

**Comment by pachulo** [comment]
Hi! I'm trying to update what I knew about this, so I will write it down to see if I got something wrong.
1. We disabled the theming app on the snap because, if it was enabled, nextcloud showed a warning about imagemagick not being available on `index.php/settings/admin/overview`.
     - This is what I remember from some years ago and it seems like it was the same some months ago [acording to this @kyrofa comment](https://github.com/nextcloud/server/issues/34338#issuecomment-1529131378).
2. It seems that the theming app is part of the nextcloud core since version 25, acording to this issue https://github.com/nextcloud/server/issues/34338
3. There seems to be interest in dropping the imagemagick dependency from the theming app, according to https://github.com/nextcloud/server/issues/36607 

And now a question:
1. It seems like I have the theming app enabled on my instance, which is running 25.0.8, because I also see the Accessibility options, but I don't see the warning about imagemagick anymore in `index.php/settings/admin/overview`, why is it so?
    - I can only see a text stating that if I want web icons I will need the PHP imagemagick extension on the foot of the `index.php/settings/admin/theming` page.

Also, take into account that I don't use the theming myself, so I might be missing somethings.

**Comment by Pilzinsel64** [comment]
I have no idea why the warning is not there for you @pachulo. For me the warning is still there.
Installing `libmagickcore-6.q16-6-extra` helped me on my two other two instanced to get rid of the warning about missing imagick php extension but the snap can't access the library probably.

Three suggestions:
- Wait until NC got rid of imagick because we don't want the warning and possible errors on users installation
- Enable now because it is a hard dependency now and we navigate users to the upstream issue (maybe pin one issue regarind the imagick warning, labled and linked to upstream, and close dups with reference to that one issue).
- Keep it disabled because don't want the warning and never want to deal with possible errors.

**Comment by scubamuc** [comment]
> I have no idea why the warning is not there for you @pachulo. For me the warning is still there.
Installing libmagickcore-6.q16-6-extra helped me on my two other two instanced to get rid of the warning about missing imagick php extension but the snap can't access the library probably.

The warning is still there on my instances too.

This has been added to [FAQ's](https://github.com/nextcloud-snap/nextcloud-snap/wiki/FAQ's#q-nextcloud-snap-shows-security--setup-warnings-what-can-i-do)

I tend to agree to suggestion two (suggestion one will be a long wait, suggestion three is like "nothing to do"):

> + Enable now because it is a hard dependency now and we navigate users to the upstream issue (maybe pin one issue regarind the imagick warning, labled and linked to upstream, and close dups with reference to that one issue).

**Comment by scubamuc** [comment]
> Installing libmagickcore-6.q16-6-extra helped me on my two other two instanced to get rid of the warning about missing imagick php extension but the snap can't access the library probably.

This didn't work for me either.

**Comment by pachulo** [comment]
> Installing libmagickcore-6.q16-6-extra helped me on my two other two instanced to get rid of the warning about missing imagick php extension but the snap can't access the library probably.

Yes, that's to be expected: snaps are isolated from the system they run, or at least that's the theory that only really applies in full to Ubuntu systems.

> Enable now because it is a hard dependency now and we navigate users to the upstream issue (maybe pin one issue regarind the imagick warning, labled and linked to upstream, and close dups with reference to that one issue).

I also vote for this option.

**Comment by pachulo** [comment]
Why did you close this @scubamuc ? 🤔

**Comment by scubamuc** [comment]
I was under the impression that its been patched already... sorry if I misunderstood

**Comment by pachulo** [comment]
What's your call here @kyrofa ?

**Comment by kyrofa** [comment]
To be honest I'm a bit confused: how are we even able to disable this at this point? Shouldn't our current logic be a complete noop? I'd like to understand that a bit better, but I'm not opposed to enabling it. We'll just need to be ready for bugs to be logged about what _still_, after all this time of our whining, looks like a warning.

**Comment by Pilzinsel64** [comment]
> how are we even able to disable this at this point?

You can disable it via command line but not via AppStore if I'm right.

> We'll just need to be ready for bugs to be logged about what still, after all this time of our whining, looks like a warning.

The warning about Imagick should be gone with next major release. If I'm right they switch the library. We then can start from zero and see if we want to install that library by default.

**Comment by pachulo** [comment]
> Shouldn't our current logic be a complete noop?

I guess we should then remove that logic, shouldn't we @kyrofa ?

> I'd like to understand that a bit better, but I'm not opposed to enabling it. We'll just need to be ready for bugs to be logged about what still, after all this time of our whining, looks like a warning.

I think it should be enough with pinning the original issue for people to be aware of. Anyway, I think we didn't have that many reports lately...or maybe my memory is failing me :sweat_smile: 

> If I'm right they switch the library. We then can start from zero and see if we want to install that library by default.

Do you have info about this @Pilzinsel64 ? I had the impression that they were going to try some [workaround for favicons in Safari and then remove the dependency](https://github.com/nextcloud/server/issues/36607)

**Comment by kyrofa** [comment]
> or maybe my memory is failing me

Or we're still actually disabling it, thus no one sees these warnings!

**Comment by Pilzinsel64** [comment]
> Do you have info about this @Pilzinsel64 ? I had the impression that they were going to try some [workaround for favicons in Safari and then remove the dependency](https://github.com/nextcloud/server/issues/36607)

You must be rith and I am wrong. Don't find auch an issue. Not sure what's wrong in my brain. 🤔 However, removing the dependency completely is probably the best solution.


> Or we're still actually disabling it, thus no one sees these warnings!

I am sure there are many people that enable the app back as they don't know the reason why it is disabled (including me some years back). But I don't know it, I just think it. ;)

**Comment by pachulo** [comment]
> Or we're still actually disabling it, thus no one sees these warnings!

Well, just did a quick manual test and it seems that this is not the case:
```
$ sudo nextcloud.occ -n app:disable theming
theming can't be disabled.
```
It seems like it's part of the core indeed.

What I cannot really understand is why I don't see the warning in my production instance, but it shows in any fresh install I do... :thinking:

**Comment by scubamuc** [comment]
@pachulo and @Pilzinsel64 
I'm not seeing this in my production logs either.
I may be wrong, but when "nextcloud-fixer" script runs on fresh install or update, the logs contain the error. See [above](https://github.com/nextcloud-snap/nextcloud-snap/pull/2535#issuecomment-1732308700)

**Comment by Pilzinsel64** [comment]
That is true. But it shouldn't break the install/update.
Anyway, let's open my PR again, I suggest.

**Comment by scubamuc** [comment]
@Pilzinsel64, nope it does't break anything I could find! Its only an error entry in the logs.

**Comment by kyrofa** [comment]
Yeah alright, let's remove that logic indeed.

**Comment by Drakonas** [comment]
So, why are this and all imagick-related issues closed (IE #1540, #2272), when the `The PHP module "imagick" is not enabled although the theming app is` error still occurs? Am I correct that this is still waiting on a major release update for the problem to be fixed? Wanted to be sure the issue is open while it is still an issue so it isn't missed.

Thank you!

**Comment by kyrofa** [comment]
This is closed because we enabled theming by default, now, which was what this issue was all about. That imagick warning is wontfix, from our perspective. @scubamuc do we have a FAQ entry for that?
